compiler/src/iree/compiler/Preprocessing/Passes.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#include "iree/compiler/Preprocessing/Passes.h"
+
+#include "iree/compiler/Preprocessing/Common/Passes.h"
+#include "mlir/Pass/PassRegistry.h"
+
+static llvm::cl::opt<std::string> clPreprocessingPassPipeline(
+    "iree-preprocessing-pass-pipeline",
```

Comment:
This is no worse than the free-standing global flags we had before but we will want to turn this into real real Options classes (to see how this is done, look at Pipelines/Options.h). Can be done as a follow-up since this isn't degrading the existing not-great state.

The reason we want to thread the options through like that is that they can be controlled from libraries, whereas the global flags are process level and we don't generally expose in a fine-grained fashion as part of APIs.

(I can clean this up and send you a PR -- it is far easier to show how to do this than explain)

Reply:
ah... let me give it a shot....

Reply:
I think I got it to work the way you suggested. PTAL.